controller/statsyncer/statsyncer.go

```diff
@@ -0,0 +1,243 @@
+			return fmt.Errorf("Error updating cluster node resources [%s] %v", cnode.Name, err)
+		}
+		// ans, _ := json.Marshal(cnode)
+		// logrus.Infof("cnode %v", string(ans))
+	}
+	return nil
+}
+
+func (s *StatSyncer) updateClusterNode(cnode *clusterv1.ClusterNode, requests map[v1.ResourceName]resource.Quantity, limits map[v1.ResourceName]resource.Quantity) error {
+	if cnode.Status.Requested == nil {
+		cnode.Status.Requested = make(map[v1.ResourceName]resource.Quantity)
+	}
+	if cnode.Status.Limits == nil {
+		cnode.Status.Limits = make(map[v1.ResourceName]resource.Quantity)
+	}
+	for name, quantity := range requests {
+		cnode.Status.Requested[name] = quantity
+	}
+	for name, quantity := range limits {
+		cnode.Status.Limits[name] = quantity
+	}
+	// ans, _ := json.Marshal(cnode.Status.Requested)
+	// logrus.Infof("cnode.Status.Requested %v", string(ans))
+	_, err := s.ClusterNodes.Update(cnode)
+	return err
+}
+
+func (s *StatSyncer) getNonTerminatedPods(nodeName string) (*v1.PodList, error) {
+	fieldSelector, err := fields.ParseSelector("spec.nodeName=" + nodeName + ",status.phase!=Succeeded" + ",status.phase!=Failed")
+	if err != nil {
+		return nil, fmt.Errorf("Skip adding cluster node resources [%s] Error getting pods %v", nodeName, err)
+	}
+	nonTerminatedPodsList, err := s.Pods.List(metav1.ListOptions{FieldSelector: fieldSelector.String()})
+	if err != nil {
+		if !errors.IsForbidden(err) {
+			return nil, fmt.Errorf("Skip adding cluster node resources [%s] Access to pods forbidden %v", nodeName, err)
+		}
+	}
+	return nonTerminatedPodsList, nil
+}
+
+func (s *StatSyncer) aggregate(data map[string]map[string]map[v1.ResourceName]resource.Quantity) (map[v1.ResourceName]resource.Quantity, map[v1.ResourceName]resource.Quantity) {
+	requests, limits := map[v1.ResourceName]resource.Quantity{}, map[v1.ResourceName]resource.Quantity{}
+	for _, podData := range data {
+		podrequests, podLimits := podData["requests"], podData["limits"]
+		for podReqName, podReqValue := range podrequests {
+			if value, ok := requests[podReqName]; !ok {
+				requests[podReqName] = *podReqValue.Copy()
+			} else {
+				value.Add(podReqValue)
+				requests[podReqName] = value
+			}
+		}
+		for podLimitName, podLimitValue := range podLimits {
+			if value, ok := limits[podLimitName]; !ok {
+				limits[podLimitName] = *podLimitValue.Copy()
+			} else {
+				value.Add(podLimitValue)
+				limits[podLimitName] = value
+			}
+		}
+	}
+	return requests, limits
+}
+
+func (s *StatSyncer) getCluster() (*clusterv1.Cluster, error) {
+	return s.Clusters.Get(s.clusterName, metav1.GetOptions{})
+}
+
+func (s *StatSyncer) getNodeMapping(nodes *v1.NodeList) map[string]*v1.Node {
+	nodeNameToNode := make(map[string]*v1.Node)
+	for _, node := range nodes.Items {
+		nodeNameToNode[node.Name] = &node
+	}
+	return nodeNameToNode
+}
+
+func isProvisioned(cluster *clusterv1.Cluster) bool {
+	isProvisioned := getConditionByType(cluster, clusterv1.ClusterConditionProvisioned)
```

Comment:
I see that we use getConditionByType and isProvisioned a lot through diff controllers. Could you move them to some sort of util method? And call from there